.github/workflows/pull_request_swift_matrix.yml

```diff
@@ -0,0 +1,84 @@
+name: Pull Request
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        type: string
+        description: "The name of the workflow used for the concurrency group."
+        required: true
+      matrix_linux_command:
+        type: string
+        description: "The command of the current Swift version linux matrix job to execute."
+        required: true
+      matrix_linux_5_8_enabled:
+        type: string
+        description: "The command of the 5.8 Swift version linux matrix job to execute."
+      matrix_linux_5_8_command_override:
+        type: string
+        description: "The command of the 5.8 Swift version linux matrix job to execute."
+      matrix_linux_5_9_enabled:
+        type: string
+        description: "The command of the 5.9 Swift version linux matrix job to execute."
+      matrix_linux_5_9_command_override:
+        type: string
+        description: "The command of the 5.9 Swift version linux matrix job to execute."
+      matrix_linux_5_10_enabled:
+        type: string
+        description: "The command of the 5.10 Swift version linux matrix job to execute."
+      matrix_linux_5_10_command_override:
+        type: string
+        description: "The command of the 5.10 Swift version linux matrix job to execute."
+      matrix_linux_nightly_next_enabled:
+        type: string
+        description: "The command of the nightly next Swift version linux matrix job to execute."
+      matrix_linux_nightly_next_command_override:
+        type: string
+        description: "The command of the nightly next Swift version linux matrix job to execute."
+      matrix_linux_nightly_main_enabled:
+        type: string
+        description: "The command of the nightly main Swift version linux matrix job to execute."
+      matrix_linux_nightly_main_command_override:
+        type: string
+        description: "The command of the nightly main Swift version linux matrix job to execute."
```

Comment:
Seems redundant. You could accept key-value pairs in a JSON.
```
fromJson(inputs.matrix_linux_command_override).510
```
assuming we have a 
```
matrix_linux_command_override:
  type: string
```
I haven't tried but it think using brackets should also work:
```
fromJson(inputs.matrix_linux_command_override)[matrix.swift_version]
```

Reply:
hmm but do you really _want_ to be backward incompatible? I think that's not a great idea to be honest.
Imagine doing a PR and noticing that CI is failing due to unrelated reasons.
I guess the maintainers can be advised to look out for these kinds of changes: "if CI failed and was complaining about an invalid value in the inputs, that's because we've probably removed it". But is that a good idea?